odoo/modules/registry.py

```diff
@@ -327,6 +327,25 @@ def field_computed(self):
                                     model_name, ", ".join(field.name for field in fields))
         return computed
 
+    @lazy_property
+    def fields_with_relational_triggers(self):
+        result = set()
+
+        for field in self.field_triggers:
+            # If the field is itself a relational field, it is also considered
+            # as triggering relational fields.
+            if field.relational or field in self.field_inverses:
+                result.add(field)
+                continue
+
+            Model = self.models[field.model_name]
+            for dep in Model._dependent_fields(field):
+                if dep in result or dep.relational or dep in self.field_inverses:
```

Comment:
The condition `dep in result` looks awkward to me.

Is it really necessary?  I personally don't think it is.

Also, what if `dep` should be in `result` but is not there yet, because the part of the algorithm that puts it in `result` is executed later?  It really looks like some sloppy transitive closure that is only correct if you execute stuff in the right order...

Reply:
The `dep in result` condition is essentially an early return statement. Meaning, it is not necessary, but improves execution speed. If `dep` is already in the result set, the other part of the condition will definitely become `True` at some point during the traversal.

> Also, what if dep should be in result but is not there yet, because the part of the algorithm that puts it in result is executed later?

This will probably happen. However, correctness is guaranteed, only optimal performance is dependent on execution order.

Ideally, I'd like to do both lazy computations of `field_triggers` and `fields_with_relational_triggers` at the same time, avoiding the extra traversal. However, I wasn't sure how to achieve this with `@lazy_property` in a readable and understandable way (it is possible by calling `setattr(self, 'fields_with_relational_triggers')` manually in the `field_triggers` computation method).